### DISCUSSION TITLE
OpenTelemetry can it retry on connection error?

### DISCUSSION THREAD

**Discussion by jimkont** [quarkusio/quarkus#36532] [Q&A] [answered]
A temporary error in the OTEL endpoint can result in closing the connection and thus, all subsequent requests will fail to export traces, until the instance is restarted. Is there a way we can configure Otel to retry connecting to the endpoint in such cases?

I observed this issue in an AWS Lambda setting with quarkus 3.4.3. A short downtime of the endpoint resulted in failing to export the traces and close the connection.  Because the Lambda instance can stay up and serve multiple requests when there is regular traffic, all the requests that are served from that instance fail to export traces. The error goes away only when that instance terminates and a new instance is started. I think this case is not limited to AWS Lambda, it can happen in a simple HTTP server setup as well.

[a comment by an automated account is not shown]

**Reply by brunobat** [reply, marked as answer]
Can you please convert this into an issue? A reproducer project would also be welcome. 
This will have to be addressed later.
